hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -134,13 +139,15 @@ private int calculateMessageDelay(long publishingMessageTimestamp) {
      * Method is calling MessageSender and is registering listeners to handle response.
      * Main responsibility of this method is that no message will be fully processed or rejected without release on semaphore.
      */
-    private void sendMessage(final Message message) {
+    private void sendMessage(final Message message, ConsumerProfiler profiler) {
         loadRecorder.recordSingleOperation();
+        profiler.measure(Measurement.ACQUIRE_RATE_LIMITER);
```

Comment:
with this invocation `stopWatch` will be stopped on a different thread than it was started on. Do we know if we can guarantee correctness in this case?

Reply:
right, the second thread might not be aware that it was started, I think that `volatile` should be enough in this case, since two threads are not changing the state of object at the same time, what do You think?